pilot/pkg/xds/delta.go

```diff
@@ -389,12 +394,17 @@ func (s *DiscoveryServer) pushDeltaXds(con *Connection,
 	t0 := time.Now()
 
 	originalW := w
-	// If subscribe is set, client is requesting specific resources. We should just generate the
+	// If delta is set, client is requesting new resources or removing old ones. We should just generate the
 	// new resources it needs, rather than the entire set of known resources.
-	if subscribe != nil {
+	// Note: we do not need to account for unsubscribed resources as these are handled by parent removal;
+	// See https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#deleting-resources.
+	// This means if there are only removals, we will send an empty response.
```

Comment:
why not skip push?

Reply:
We do actually, I need to update the comment